### DISCUSSION TITLE
How to update ISCE results with new Sentinel-1 epochs

### DISCUSSION THREAD

**Discussion by SMJMirza** [isce-framework/isce2#590] [Q&A]
Hi all,

I produced the ISCE products for 20 Sentinel-1 ascending and descending frames on a large scale and then tried to do time series processing using the MintPy. Now, my challenge is how I can update the results of these frames with the new Sentinel-1 epochs. As you know, the ISCE workflow includes 16 run_files to complete the InSAR progress (as attached here), but should I run all steps for the new epochs, especially the run_files for the co-registration steps (run_01 to run_12)?

Since MintPy does not need the large files generated in the first steps of ISCE (run_01 to run_13), we can remove them to save storage, but I am worried if I need to keep them for updating by the new Sentinel-1 epochs.

Sincerely,
Sayyed

[run_all.txt](https://github.com/isce-framework/isce2/files/9639121/run_all.txt)

**Comment by manudeo** [comment]
Hi Sayyed,

To extend an existing network, there are two ways that I implement - 

1. Keep all files. In this case, when you add new SLCs and run the stackSentinel again, it will regenerate run_files by skipping already existing processing and containing recipes for only new SLCs. In this case, run 01 will be entirely escaped (remember to use the same reference date).  New run files will have run_01 to run_15 (instead of run 16).
2. Keep only last (c-1) intermediate files. Here c is the number of connections, if you are using 3 connections, keep all intermediate files and SLCs belonging to last two dates of your time-series. Keep reference and geom_reference folders and everything inside merge folder as it is. Add new SLCs and rerun stackSentinal. It will again make 15 run files, skipping the first run. For simplicity, one can even move coreg_secondarys and secondarys in a new folder and then run stackSentinel. In this case, stackSentinel will produce 16 run files, skip run_01, move back the corge_secondarys and secondarys folders in the working directory, start from run_02. 

After finishing run_16 (or run_15, if using method 1), merged folder will have new dates. Rerun mintpy, it will update new dates.

**Comment by SMJMirza** [comment]
Hi Manudeo,

Thank you for sharing your experience updating the ISCE results with the new Sentinel-1 SLCs.
I will try to see which one can work well for me.

Sincerely,
Sayyed

**Comment by Karthi-Wolf** [comment]
Hi Mr. Manudeo Singh,
I need to do this on a coregistration workflow. 
I have done coregistration upto 2025. Now I need to add the 2026 SLC images to the stack. 
How can I do this?

Regards.

**Reply by manudeo** [reply]
Same workflow applies.
